core/drand_beacon.go

```diff
@@ -69,108 +71,59 @@ type Drand struct {
 	version utils.Version
 }
 
-// NewDrand returns a drand struct. It assumes the private key pair
-// has been generated and saved already.
-func NewDrand(s key.Store, c *Config) (*Drand, error) {
-	d, err := initDrand(s, c)
-	if err != nil {
-		return nil, err
-	}
-	return d, nil
-}
+func NewBeaconProcess(log log.Logger, version utils.Version, store key.Store,
```

Comment:
Ideally we would like that to be a generic interface - we dont know what kind of protocols we want to run and we want maximum isolation: the daemon should only be responsible for loading them and dispatching msg, and that's it - currently he's doing some logic of the beaconprocess itself (like deciding if it has launched or not, instantiating a key store from the folder etc).
Given the timeframe I am ok to continue like this for this PR but please let's make an issue and resolve it after this PR is merged.

Reply:
This was partially fixed on a previous commit. The issue #860 address this.